code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/OptimizeProposition.java

```diff
@@ -153,10 +206,20 @@ public static OptimizeProposition fromEventData(final Map<String, Object> data)
                 return null;
             }
 
-            final Map<String, Object> scopeDetails =
+            // Get existing scopeDetails or create new one
+            Map<String, Object> scopeDetails =
                     DataReader.getTypedMap(
                             Object.class, data, OptimizeConstants.JsonKeys.PAYLOAD_SCOPEDETAILS);
 
+            // Parse activity and placement objects
+            Map<String, Object> activity =
+                    DataReader.getTypedMap(
```

Comment:
`DataReader.getTypedMap` will throw an exception is the key is not found. Since scope details, activity and placement are optional field will this fail if those fields don't exist? You can use `DataReader.optTypedMap` in that case

Reply:
Same comment for L211 and L220